src/app/hooks/useSearchQueryNetworkParam.ts

```diff
@@ -0,0 +1,29 @@
+import { useSearchParams } from 'react-router-dom'
+import { Network } from '../../types/network'
+import { RouteUtils } from '../utils/route-utils'
+
+/**
+ * Get the network from the URL search params and return it if it is valid,
+ * otherwise fallback to mainnet
+ */
+export const useSearchQueryNetworkParam = (): {
+  network: Network
+  setNetwork: (network: Network) => void
+} => {
+  const [searchParams, setSearchParams] = useSearchParams()
+  let networkQueryParam = searchParams.get('network')
+  if (!networkQueryParam || !RouteUtils.getEnabledNetworks().includes(networkQueryParam as any)) {
+    networkQueryParam = Network.mainnet
+  }
+  return {
+    network: networkQueryParam as Network,
+    setNetwork: network => {
+      if (network === Network.mainnet) {
+        searchParams.delete('network')
+        setSearchParams(searchParams)
+      } else {
+        setSearchParams({ network })
+      }
```

Comment:
Slightly more accurate:
```suggestion
      if (network === Network.mainnet) {
        searchParams.delete('network')
      } else {
        searchParams.set('network', network)
      }
      setSearchParams(searchParams)
```

(if we ever used it with other query parameters)